lib/semantics/resolve-names.cc

```diff
@@ -3969,18 +3974,25 @@ bool ConstructVisitor::Pre(const parser::LocalitySpec::LocalInit &x) {
   }
   return false;
 }
+
 bool ConstructVisitor::Pre(const parser::LocalitySpec::Shared &x) {
-  for (auto &name : x.v) {
-    if (auto *prev{FindSymbol(name)}) {
-      if (prev->owner() == currScope()) {
-        SayAlreadyDeclared(name, *prev);
-      }
-      auto &symbol{MakeSymbol(name, HostAssocDetails{*prev})};
-      symbol.set(Symbol::Flag::LocalityShared);
-    } else {
+  for (const auto &name : x.v) {
+    auto *prev{FindSymbol(name)};
+    if (!prev) {
       Say(name, "Variable '%s' not found"_err_en_US);
       context().SetError(
           MakeSymbol(name, ObjectEntityDetails{EntityDetails{}}));
+    } else {
+      if (prev->owner() == currScope()) {
+        SayAlreadyDeclared(name, *prev);  // C1125 and C1126
+      } else {
+        if (!IsVariableName(*prev)) {
+          SayBadLocality(name, *prev);  // C1124
+        } else {
+          auto &symbol{MakeSymbol(name, HostAssocDetails{*prev})};
+          symbol.set(Symbol::Flag::LocalityShared);
+        }
+      }
```

Comment:
I think that this kind of cascading `if` is better written like this:
```
    if ...
    } else if ...
    } else if ...
    } else ...
    }
```

Reply:
Thanks, Tim.  I'll fix this.